tests/components/knx/test_diagnostic.py

```diff
@@ -29,70 +29,71 @@
 from tests.typing import ClientSessionGenerator
 
 
+@pytest.mark.parametrize("yaml_config", [{}])
 async def test_diagnostics(
     hass: HomeAssistant,
     hass_client: ClientSessionGenerator,
     mock_config_entry: MockConfigEntry,
     knx: KNXTestKit,
+    mock_yaml_configuration: None,
 ) -> None:
     """Test diagnostics."""
     await knx.setup_integration({})
```

Comment:
The parameter here is actually the yaml config passed.
https://github.com/home-assistant/core/blob/b11539fb0415c1636ddd3a453e56a44ca6b0bc27/tests/components/knx/conftest.py#L91
Would it be possible to retrieve this from the mock in case it would be needed in the patched test? 

In these tests it doesn't really matter as we are using `knx.setup_integration` just to mock the config entry and yaml is read fresh by `async_get_config_entry_diagnostics` to be able to have diagnostics even with invalid yaml.
https://github.com/home-assistant/core/blob/b78318a6176ae11221a626d3e555375828330237/homeassistant/components/knx/diagnostics.py#L43

Reply:
Of course, you can add it to the function parameters.

Reply:
Ah, I see. So we can use 
```py
@pytest.mark.parametrize("hass_config", [{}])
async def test_test(
    hass: HomeAssistant,
    mock_yaml_configuration: None,
    hass_config: ConfigType | None,
 ) -> None:
    await async_setup_component(hass, DOMAIN, hass_config) 
 ```
Thanks for clarification!